src/component/select/select.component.ts

```diff
@@ -1246,7 +1245,7 @@ export class DejaSelectComponent extends ItemListBase implements ControlValueAcc
                 if (this.modelIsValue) {
                     const valueField = this.getValueField();
                     if (models.find((m) => !!m[valueField])) {
-                        output = models.map((m) => m[valueField] || m);
+                        output = models.map((m) => m[valueField] !== undefined ? m[valueField] : m);
```

Comment:
Why not using "m[valueField] === undefined ? m : m[valueField] );" instead to avoid a negative test